core/store/src/trie/iterator.rs

```diff
@@ -716,39 +343,4 @@ mod tests {
             .collect();
         assert_eq!(got, want);
     }
-
-    #[test]
-    fn test_has_value() {
```

Comment:
Can be migrated to `TrieIteratorImpl for DiskTrieIteratorInner`. Maybe there is no value in keeping it, but there is still a need to get all state KV pairs for state part AFAIU.

Reply:
I mean, this test didn't make too much sense to me. It's basically checking if an internal implementation detail `IterStep` is of type value, then value should exist, else it shouldn't?

I think this test was originally in place as we had the function `has_value()` exposed from the iterator which no longer exists, hence the test is also irrelevant.

If we care about this for state parts, the better way of testing this is just doing the actual iteration and checking whether all data that we want exists.

Reply:
It checked whether two different ways to check value existence match each other. However, your argument makes sense.